From c2e47515b7ca9d6398105888063d1aa0a18508d3 Mon Sep 17 00:00:00 2001
From: Greg Ungerer <gerg@kernel.org>
Date: Tue, 26 Apr 2022 12:14:24 +1000
Subject: [PATCH] uClibc-ng: add ELF-FDPIC support for armnommu

Add support for ELF-FDPIC executables to the arm startup code.
This is really just ELF support not full FDPIC support.

The relocation (link/loading) is done in a separate uld.so.1 (ELF
interpreter) binary. Just a few tweaks in here to get the stack
arguments right for the slight differences with the ELF-FDPIC loader.

Signed-off-by: Greg Ungerer <gerg@kernel.org>
---
 libc/sysdeps/linux/arm/crt1.S |   1 +-
 1 file changed, 1 insertions(+), 1 deletions(-)

diff --git a/libc/sysdeps/linux/arm/crt1.S b/libc/sysdeps/linux/arm/crt1.S
--- a/libc/sysdeps/linux/arm/crt1.S	2023-01-26 00:33:56.187322935 +1000
+++ b/libc/sysdeps/linux/arm/crt1.S	2023-01-26 00:34:10.883426967 +1000
@@ -245,7 +245,7 @@
 	mov fp, #0
 	mov lr, #0
 
-#ifdef __ARCH_USE_MMU__
+#if defined(__ARCH_USE_MMU__) || defined(FDPIC_LOADER)
 #ifdef L_rcrt1
 	/* We don't need to save a1 since no dynamic linker should have run */
 	ldr a1, .L_GOT          /* Get value at .L_GOT + 0  (offset to GOT)*/
-- 
2.25.1
